src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -308,12 +314,14 @@ def update_app_settings(resource_group_name, name, settings=None, slot=None, slo
         # split at the first '=', appsetting should not have '=' in the name
         settings_name, value = name_value.split('=', 1)
         app_settings.properties[settings_name] = value
+    client = web_client_factory()
+    if slot is None:
+        result = client.web_apps.update_application_settings(resource_group_name, name, str, app_settings.properties)  # pylint: disable=line-too-long
+    else:
+        result = client.web_apps.update_application_settings_slot(resource_group_name, name, slot, str, app_settings.properties)  # pylint: disable=line-too-long
```

Comment:
I would suggest keep on using `_generic_site_operation`.

Reply:
It looks like '_generic_site_operation' only takes one extra parameter. I need two in this case.

Reply:
can you improve the `_generic_site_operation`? You have 4+ places where you can leverage to consolidate the code

Reply:
I think using client.web_apps.operation(..) is fine.

Reply:
We are having 5 people working on this file. Getting code simple and consolidated is a good thing for maintenance.
Hence I suggest the following code snippets
```python
def _generic_settings_operation(resource_group_name, name, operation_name, setting_properties, slot=None, client=None):
     client = client or web_client_factory()
     m = getattr(client.web_apps, operation_name if slot is None else operation_name + '_slot')
     if slot is None:
         return m(resource_group_name, name, str, setting_properties)
     else:
        return m(resource_group_name, name, slot, str, setting_properties)

# then caller will be simplified like
 _generic_settings_operation(resource_group_name, name, 'update_application_settings', app_settings.properties, slot, client)
# or
 _generic_settings_operation(resource_group_name, name, 'update_connection_strings', conn_strings.properties, slot, client)
```